src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -16,26 +22,41 @@
     private String name;
     private String color;
 
-    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
-    private List<Section> sections = new ArrayList<>();
+    @Embedded
+    private Sections sections = new Sections();
 
-    public Line() {
+    public void addSection(Section section) {
+        sections.add(section);
+        section.setLine(this);
     }
 
-    public Line(String name, String color) {
+    protected Line() {
+
+    }
+
+    private Line(Long id, String name, String color, Section section) {
+        this.id = id;
         this.name = name;
         this.color = color;
+
+        addSection(section);
+    }
+
+    public static Line of(String name, String color, Section section) {
+        return new Line(null, name, color, section);
     }
 
-    public Line(String name, String color, Station upStation, Station downStation, int distance) {
+    public static Line of(Long id, String name, String color, Section section) {
+        return new Line(id, name, color, section);
+    }
+
+    public void update(String name, String color) {
         this.name = name;
         this.color = color;
-        sections.add(new Section(this, upStation, downStation, distance));
     }
 
-    public void update(Line line) {
-        this.name = line.getName();
-        this.color = line.getColor();
+    public void removeSectionByStation(Station station) {
```

Comment:
```suggestion
    public void removeSectionBy(Station station) {
```
파라미터를 보면 station이 필요한 것을 알 수 있지 않을까요